src/BatchProcessDialog.cpp

```diff
@@ -1303,9 +1303,12 @@ void MacrosWindow::OnDown(wxCommandEvent & WXUNUSED(event))
 
 void MacrosWindow::OnApplyToProject(wxCommandEvent & event)
 {
+   AudacityProject *project = &mProject;
+   project->mApplyToProject = true;
```

Comment:
You did NOT follow the example of MacroCommands::ApplyCommandInBatchMode in the use of finally() to reset the flag back to false.

There is a very important difference between that, and what you wrote here.  With finally() the restoring of the state is guaranteed even if the applying of the command throws and exception.  Not so as written here.

Reply:
If you instead use ProjectFileIO::Get(mProject) to get the object, then you might need to set the flag in it and set it again, with a setting method paired with the new accessor method.

But don't to that.  Do a more complicated but better thing.  See class ValueRestorer.  Give ProjectFileIO a method like

```
ValueRestorer<bool> ProjectFileIO::ApplyingToProjectScope()
```

and at point of use

```
auto scope = ProjectFileIO::ApplyingToProjectScope();
```

And then you don't need a finally(), but scope serves the purpose of guaranteeing cleanup when it is destroyed.

The reason for this trouble?  Because making a new variable with a getter and a setter is little better than just making a public variable.  But instead you can constrain the changes of state, so that users of ProjectFileIO cannot to otherwise but change the variable in this safer way with guaranteed cleanup.

Reply:
If you don't know the C++ programming pattern called "RAII," please look it up and learn it!  What I am recommending here falls under the wider patter.   It is applied, not to recover memory or another resource, but rather to guarantee that temporary changes that a function makes to its environment (the state of ProjectFileIO here) are guaranteed to be undone no matter how the function may exit.